### PR TITLE
Add first rule for sulu rector

### PR DESCRIPTION
**Pull request by alexander-schranz**
This pull request should contain the first rule which should upgrade the deprecated methods on the https://github.com/sulu/sulu/blob/2.5/src/Sulu/Component/Localization/Localization.php object.

### CODE REVIEW

**Comment by alexander-schranz** [comment]
Seems like I'm currently missing something in the test setup. I run into:

```
InvalidArgumentException: Could not resolve argument "Rector\Config\RectorConfig $rectorConfig" for
"/home/runner/work/sulu-rector/sulu-rector/vendor/rector/rector-src/src/Kernel/../../confi/config.php".
...
Caused by
LogicException: You cannot use the ConfigBuilders without providing a class implementing ConfigBuilderGeneratorInterface.
```

/cc @tomasvotruba

Bildschirmfoto 2022-05-12 um 16 20 18

<details><summary>Quick Setup Script if you want try locally</summary>

```bash
git clone git@github.com:alexander-schranz/sulu-rector.git
cd sulu-rector/
git checkout feature/first-rule
composer update
composer test
```

</details>

**Comment by TomasVotruba** [comment]
My rought guess would be:

```bash
composer install
```

Why? When you use `"rector/rector-src": "dev-main",` in `composer.json`, the patches are not applied yet. So you have to "compile" Rector yourself.

I recommend using "rector/rector" as standard and downgraded to PHP 7.1. The "rector/rector-src" is only for core Rector packages and require PHP 8.1

**Comment by TomasVotruba** [comment]
The best inspiration of 3rd party using Rector for upgrades would be:
https://github.com/craftcms/rector/blob/main/composer.json

**Comment by alexander-schranz** [comment]
@TomasVotruba Thx for the quick response and thx for the reference project. Was using `rector-symfony` as reference, but seems like there it works differently. The tests are running now :) Now I need to checkout why the ruleset is not yet applied correctly.

**Comment by TomasVotruba** [comment]
:+1: 

I'd like to run it locally myself, but fork and PR does not work quote easily :/ 
Could you merge this one to `master` or give me contributors access here?

File: tests/Set/Sulu24/Fixture/localization.php.inc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        $localization->isXDefault();
+    }
+}
+
+?>
+---
+<?php
+
+class SomeClass {
+    public function someMethod(\Sulu\Component\Localization\Localization $localization)
```
Comment (TomasVotruba) [inline tests/Set/Sulu24/Fixture/localization.php.inc:17]: Does this class exist in "require-dev" dependencies of `composer.json` or local `/stubs` directory?

Without existing class, PHPStan cannot find it nor analyse it.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 # composer
 /composer.phar
-/composer.lock
```
Comment (TomasVotruba) [inline .gitignore:3]: I think you can keep this one there. This is just a tooling package, so maintaining dependencies are extra work that will not pay off.

Also, the upgrade tool should always be tested on bleeding edge versions, so the upgrade to the latest version works :)
